### PR TITLE
String.compareTo for AArch64

### PR DESCRIPTION
**Pull request by dsamersoff**
String compareTo intrinsic for  AArch64

### CODE REVIEW

**Comment by dsamersoff** [comment]
Style check failed with strange warning from findbugs that I can't reproduce locally. So leaving it as is and address it later.

File: compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64ArrayCompareToOp.java [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+    private final int array1BaseOffset;
+    private final int array2BaseOffset;
+
+    @Def({REG}) protected Value resultValue;
+
+    @Alive({REG}) protected Value array1Value;
+    @Alive({REG}) protected Value array2Value;
+    @Alive({REG}) protected Value length1Value;
+    @Alive({REG}) protected Value length2Value;
```
Comment (dougxc) [inline compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64ArrayCompareToOp.java]: Please check with @mur47x111 whether the [fix](https://github.com/oracle/graal/commit/c5e87525231707e6c0804b97a72cf90836910c4f#diff-28a415c4cd88eb4a9c38354ecb216c17) he made for GR-8748 is also needed here.

File: compiler/src/org.graalvm.compiler.replacements.test/src/org/graalvm/compiler/replacements/test/StringCompareToTest.java [hunk 2]
```diff
@@ -100,15 +102,29 @@ public void testEqualString() {
     @Test
     @Ignore("GR-8748")
```
Comment (dougxc) [inline compiler/src/org.graalvm.compiler.replacements.test/src/org/graalvm/compiler/replacements/test/StringCompareToTest.java]: I don't think we need this `@Ignore` any more right @mur47x111 ?

Comment (mur47x111) [reply compiler/src/org.graalvm.compiler.replacements.test/src/org/graalvm/compiler/replacements/test/StringCompareToTest.java]: My commit has removed this annotation and reenable the test.

File: compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64ArrayCompareToOp.java (hunk 1 above)
Comment (mur47x111) [reply compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64ArrayCompareToOp.java]: Please apply the same pattern as [here](https://github.com/oracle/graal/commit/a188baff1435dd34a96f76d21275e269cfa546b3#diff-28a415c4cd88eb4a9c38354ecb216c17) to ensure modifiable variables are killed after the intrinsic.

**Comment by dsamersoff** [comment]
Adopted all comments. The fix is ready to push.

**Pull request merged**